lib/cypress/population_clone_job.rb

```diff
@@ -100,6 +102,33 @@ def unnumerify(patient)
       end
     end
 
+    def restrict_entry_codes(cloned_patient)
+      # Loop through every data element
+      cloned_patient.qdmPatient.dataElements.each do |entry|
+        # look up if the vendor has a preferred_code_system for the qdmCategory
+        vendor_preference = @test.product.vendor.preferred_code_systems[entry.qdmCategory]
+        # if the vendor does not have a preference, pick the first code
+        if vendor_preference.blank?
+          entry.dataElementCodes = [entry.dataElementCodes[0]]
+        else
+          # if the vendor has a preference, loop through in order
+          vendor_preference.each do |vp|
```

Comment:
Quick note that *if* there's a vendor preference, it should contain all potential code systems, such that there will always be *some* option for the code choice (i.e. don't want a preference for just CPT and then to come up empty when there's only a SNOMED code available)

Reply:
Could make a code change to make this safer or I think could just add a note/comment somewhere for when the "preferred_code_systems" hash is created.